examples/hash_cracker_ray.py

```diff
@@ -0,0 +1,142 @@
+import argparse
+import itertools
+import math
+from datetime import datetime
+from hashlib import sha256
+from typing import Optional
+
+import ray
+
+# some sample phrases to crack
+#
+# 'G0!3m':   070063ad89872aaa0fc0a2f170be5641e9c5d201d25b76703a4be3ee3848016c
+# 'golem':   4c5cddb7859b93eebf26c551518c021a31fa0013b2c03afa5b541cbc8bd079a6
+# '9Lm!':    de6c0da53ac2bf2b6954e400767106011e4471db7a412cce0388e3441e0ad2ec
+# `test`:    9f86d081884c7d659a2feaa0c55ad015a3bf4f1b2b0b822cd15d6c15b0f00a08
+# `foo`:     2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae
+# `glm`:     2bf548d8056029c73e6e28132d19a3a277a49daf32b1c1ba7b0b7fc7e78bf5cd
+# `be`:      46599c5bb5c33101f80cea8438e2228085513dbbb19b2f5ce97bd68494d3344d
+# `x`:       2d711642b726b04401627ca9fbac32f5c8530fb1903cc4db02258717921a4881
+
+# the character table that we want to use to construct possible phrases
+CHARS = [
+    chr(c)
+    for c in itertools.chain(
+        range(ord("a"), ord("z") + 1),
+        range(ord("A"), ord("Z") + 1),
+        range(ord("0"), ord("9") + 1),
+        range(ord(" "), ord("/") + 1),
+        range(ord(":"), ord("@") + 1),
+        range(ord("["), ord("`") + 1),
+        range(ord("{"), ord("~") + 1),
+    )
+]
+
+start_time = datetime.now()
```

Comment:
I would move this closer to main execution loop

Reply:
oops, good catch :sweat_smile: